tests/integration-tests/tests/proxy/test_proxy.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import boto3
+import pytest
+from assertpy import assert_that
+from cfn_stacks_factory import CfnStack
+from remote_command_executor import RemoteCommandExecutor
+from utils import generate_stack_name, get_username_for_os
+
+
+@pytest.fixture(scope="class")
+def proxy_stack(region, request, cfn_stacks_factory):
+    """
+    Set up and tear down a CloudFormation stack to deploy a proxy environment.
+
+    The stack is created from a CloudFormation template that sets up the necessary
+    networking, instances, and security groups for a proxy environment. The proxy
+    address and subnet are retrieved from the stack outputs and used in the cluster
+    configuration.
+    """
+    proxy_stack_template_path = "../../cloudformation/proxy/proxy.yaml"
+    with open(proxy_stack_template_path) as proxy_stack_template:
+        stack_name = generate_stack_name("integ-tests-proxy", request.config.getoption("stackname_suffix"))
+        stack_parameters = [
+            {"ParameterKey": "Keypair", "ParameterValue": request.config.getoption("key_name")},
+            {"ParameterKey": "VpcCidr", "ParameterValue": "10.0.0.0/16"},
+        ]
+        capabilities = ["CAPABILITY_IAM"]
+        tags = [{"Key": "parallelcluster:integ-tests-proxy-stack", "Value": "proxy"}]
+
+        proxy_stack = CfnStack(
+            name=stack_name,
+            region=region,
+            template=proxy_stack_template.read(),
+            parameters=stack_parameters,
+            capabilities=capabilities,
+            tags=tags,
+        )
+
+    cfn_stacks_factory.create_stack(proxy_stack)
+    yield proxy_stack
+
+    if not request.config.getoption("no_delete"):
+        cfn_stacks_factory.delete_stack(proxy_stack.name, region)
+
+
+def get_instance_public_ip(instance_id, region):
+    ec2_client = boto3.client("ec2", region_name=region)
+    response = ec2_client.describe_instances(InstanceIds=[instance_id])
+    instance = response["Reservations"][0]["Instances"][0]
+    return instance.get("PublicIpAddress")
+
+
+@pytest.mark.usefixtures("region", "os", "instance", "scheduler")
+def test_proxy(pcluster_config_reader, clusters_factory, proxy_stack, scheduler_commands_factory):
+    """
+    Test the creation and functionality of a ParallelCluster using a proxy environment.
```

Comment:
**minor**
"of a ParallelCluster" = "of a cluster"

Reply:
Done!